### PR TITLE
conformance: fix ipv6 addresses

### PR DESCRIPTION
**Pull request by howardjohn**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:
This was a regression in 0.7.0-rc2. We generated invalid IPv6 address

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Fixed an issues causing conformance tests to fail when using IPv6 addresses
```

### CODE REVIEW

File: conformance/utils/http/http.go [hunk 1]
```diff
@@ -148,12 +148,21 @@ func calculateHost(gwAddr, scheme string) string {
...
 	}
-	return host + ":" + port
+	return gwAddr
+}
+
+func ipv6SafeHost(host string) string {
+	// We assume that host is a literal IPv6 address if host has
+	// colons.
+	if strings.Contains(host, ":") {
+		return "[" + host + "]"
```
Comment (aojea) [inline conformance/utils/http/http.go:163]: but you only maybe you should reference this https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2?

**Comment by robscott** [comment]
Thanks @howardjohn!

/lgtm
/approve

**Pull request merged**